### DISCUSSION TITLE
How should I approach building a personal portfolio/blog website?

### DISCUSSION THREAD

**Discussion by Ashvith10** [vanjs-org/van#375] [General]
I have previously created a personal portfolio in SvelteKit where I was able to make use of both SSG and SSR using [prerender](https://kit.svelte.dev/docs/page-options#prerender). I wish to have some part of the blog use JavaScript for theming and simple animation (like hamburger menu, page transition, etc), but I also want the blog part (which will probably use Markdown) to be transpiled to HTML. Would appreciate discussions on what tools and libraries I should go with.

**Comment by Tao-VanJS** [comment]
FWIW, we have the tool and library that convert MD/HTML strings into valid VanJS statements. You can refer to the "HTML/MD to VanJS" page of VanJS website.

**Reply by Ashvith10** [reply]
@Tao-VanJS thank you for responding. I was looking for a list of tools that I should be looking into, but after a little bit of research here and there, I'm assuming that Mini-Van is what I'll be needing, alongside a server-side framework?

You've also mentioned [convertor](https://github.com/vanjs-org/converter) - does it pre-compile to Markdown while building, or will this be done on-the-go?

**Reply by Tao-VanJS** [reply]
Hi @Ashvith10, 

The converter library you mentioned can convert a markdown string into valid JavaScript code that calls VanJS to build the corresponding DOM tree. Thus it's appropriated to be used at the build-time of your application.

**Comment by sirenkovladd** [comment]
oh, this is a very interesting discussion, I've asked myself this question many times

As a result, I realized that the most convenient option for me is to use [Bun.macros](https://bun.sh/docs/bundler/macros) for SSR / SSG
and just reuse the components for CSR where needed

**Reply by sirenkovladd** [reply]
This repo is still just a draft, but you can take a look here
https://github.com/sirenkovladd/rss-notification
